source/postprocess/mobility_statistics.cc

```diff
@@ -0,0 +1,157 @@
+                {
+                  fe_face_values.reinit (cell, f);
+                  // extract velocities
+                  fe_face_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
+                      surface_velocity_values);
+
+                  // determine the squared velocity on the face
+                  double local_sqvel = 0.0;
+                  double local_fe_face_area = 0.0;
+                  for (unsigned int q = 0; q < n_q_points; ++q)
+                    {
+                      const double JxW = fe_face_values.JxW(q);
+                      local_sqvel += surface_velocity_values[q].norm_square() * JxW;
+                      local_fe_face_area += JxW;
+                    }
+                  local_velocity_square_integral_top_boundary += local_sqvel;
+                  local_top_boundary_area += local_fe_face_area;
+                }
+          }
+
+      // now communicate to get the global values and convert to rms
+      const double global_velocity_square_integral
+        = Utilities::MPI::sum (local_velocity_square_integral, this->get_mpi_communicator());
+
+      const double global_rms_vel = std::sqrt(global_velocity_square_integral) /
+                                    std::sqrt(this->get_volume());
+
+      const double global_top_velocity_square_integral = Utilities::MPI::sum(local_velocity_square_integral_top_boundary, this->get_mpi_communicator());
+      const double global_top_boundary_area_integral = Utilities::MPI::sum(local_top_boundary_area, this->get_mpi_communicator());
+
+      const double global_top_rms_vel = std::sqrt(global_top_velocity_square_integral) /
+                                        std::sqrt(global_top_boundary_area_integral);
+
+      // now add the computed rms velocities to the statistics object
+      // and create a single string that can be output to the screen
+      const double mobility  = global_top_rms_vel/global_rms_vel;
+      const std::string name_mobility = "Mobility";
+
+      statistics.add_value (name_mobility,
+                            mobility );
+
+      // also make sure that the other columns filled by this object
+      // all show up with sufficient accuracy and in scientific notation
+      statistics.set_precision (name_mobility, 8);
+      statistics.set_scientific (name_mobility, true);
+
+      std::ostringstream output;
+      output.precision(3);
+      output << mobility;
+
+      return std::pair<std::string, std::string> ("Mobility:",
+                                                  output.str());
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(MobilityStatistics,
+                                  "mobility statistics",
+                                  "A postprocessor that computes some statistics about the "
+                                  "mobility of the lithosphere.")
```

Comment:
Try to keep this description in sync with what you write at the top of the header file. It is a bit  duplicative, but the text in the header file will end up in our online documentation (https://aspect.geodynamics.org/doc/doxygen/annotated.html), while the text here will end up in the manual and the parameter website (https://aspect.geodynamics.org/doc/parameter_view/parameters.xml). Ideally both should contain the same information, in particular the references to the papers.

Reply:
Ok, thanks